src/kpatch_ptrace.c

```diff
@@ -1162,14 +1164,21 @@ int kpatch_ptrace_attach_thread(kpatch_process_t *proc,
 			status = WTERMSIG(status);
 
 
-		ret = ptrace(PTRACE_CONT, pctx->pid, NULL,
+		ret = ptrace(PTRACE_CONT, tid, NULL,
 			     (void *)(uintptr_t)status);
 		if (ret < 0) {
 			kplogerror("can't cont tracee\n");
 			return -1;
 		}
 	} while (1);
 
+	pctx = kpatch_ptrace_ctx_alloc(proc);
```

Comment:
In case of failure here you leave the thread attached but unaccounted. Please fix restore the allocations before the attachment.

Reply:
Moved pctx allocation before SIGSTOP processing, but after successfull PTRACE_ATTACH.